archimedes/function.py

```diff
@@ -0,0 +1,42 @@
+from errors import ArchimedesError
+from scope import Scope
+
+class Function:
+    def __init__(self, parser, arg_defs, production):
+        self.arg_defs = arg_defs
+        self.production = production
+        self.parent_scope = parser.current_scope
+        self.parser = parser
+
+    def getargs(self, args):
```

Comment:
Hmm.  Is this necessary?  I feel like we can use python itself to do the heavy lifting of expanding args, and buy ourselves more flexibility, too.

What if calls are formatted like this:
```
{
  args: ["a", "b"],
  kwargs: {"c": "d"}
}
```

In that case, we can simply "invoke" a function `set_args(*payload['args'] or [], **parload['kwargs'] or {})` or even just `call(...)` and allow us to specify arguments using natural python syntax and gain it's full calling convention.  

If we want to convert errors to archimedes errors, we can also broadly catch the `TypeError` and convert that?

But just minor suggestion.